From 39873560778eb1bd75cabc5ef1c9a155e4491de7 Mon Sep 17 00:00:00 2001
From: Luis Ponce <luis.f.ponce.navarro@linux.intel.com>
Date: Tue, 11 Sep 2018 14:02:24 -0500
Subject: [PATCH 1/2] Allow Clear Linux detection in python2 and python3

The version.py has three variables where stores:

*In case it is run in python2 and by using platform.linux_distribution
 + DISTRO_NAME      = os-release -> NAME
 + DISTRO_VERSION   = os-release -> VERSION
 + DISTRO_FULL_NAME = os-release -> NAME

*In case it is run in python3.7 and by using distro.linux_distribution
(since platform.linux_distribution deprecated according to
https://bugs.python.org/issue1322)
 + DISTRO_NAME      = os-release -> ID
 + DISTRO_VERSION   = os-release -> VERSION
 + DISTRO_FULL_NAME = os-release -> NAME

Due the possibility of Clear Linux NAME modification it's been decided
to look only for "Clear Linux" in the DISTRO_FULL_NAME string,
so in this manner both scenarios above will work as expected.
---
 azurelinuxagent/common/osutil/factory.py  | 3 +--
 azurelinuxagent/pa/deprovision/factory.py | 2 +-
 setup.py                                  | 3 +--
 3 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/azurelinuxagent/common/osutil/factory.py b/azurelinuxagent/common/osutil/factory.py
index 223c3a2..82271fc 100644
--- a/azurelinuxagent/common/osutil/factory.py
+++ b/azurelinuxagent/common/osutil/factory.py
@@ -44,8 +44,7 @@ def get_osutil(distro_name=DISTRO_NAME,
     if distro_name == "arch":
         return ArchUtil()
 
-    if distro_name == "clear linux os for intel architecture" \
-            or distro_name == "clear linux software for intel architecture":
+    if "Clear Linux" in distro_full_name:
         return ClearLinuxUtil()
 
     if distro_name == "ubuntu":
diff --git a/azurelinuxagent/pa/deprovision/factory.py b/azurelinuxagent/pa/deprovision/factory.py
index 21169b3..07c26bf 100644
--- a/azurelinuxagent/pa/deprovision/factory.py
+++ b/azurelinuxagent/pa/deprovision/factory.py
@@ -40,7 +40,7 @@ def get_deprovision_handler(distro_name=DISTRO_NAME,
             return UbuntuDeprovisionHandler()
     if distro_name == "coreos":
         return CoreOSDeprovisionHandler()
-    if distro_name == "clear linux":
+    if "Clear Linux" in distro_full_name:
         return ClearLinuxDeprovisionHandler()
 
     return DeprovisionHandler()
diff --git a/setup.py b/setup.py
index 063779a..624651c 100755
--- a/setup.py
+++ b/setup.py
@@ -107,8 +107,7 @@ def get_data_files(name, version, fullname):
         set_udev_files(data_files)
         set_files(data_files, dest="/usr/share/oem",
                   src=["init/coreos/cloud-config.yml"])
-    elif name == 'clear linux os for intel architecture' \
-            or name == 'clear linux software for intel architecture':
+    elif "Clear Linux" in fullname:
         set_bin_files(data_files, dest="/usr/bin")
         set_conf_files(data_files, dest="/usr/share/defaults/waagent",
                        src=["config/clearlinux/waagent.conf"])
-- 
2.20.0

